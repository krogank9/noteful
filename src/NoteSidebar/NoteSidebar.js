import React from 'react';
import { withRouter } from 'react-router';

import NotefulContext from '../NotefulContext.js';

class NoteSidebar extends React.Component {
	static contextType = NotefulContext;
	
	render() {
		let noteId = this.props.match.params.noteId;
		let note = this.context.getNoteFromId(noteId);
		let folderId = note && note.folderId;
		let folder = this.context.getFolderFromId(folderId);
		
		return (
			<div>
				<button onClick={() => this.props.history.goBack()}>Go back</button>
				<h2>Folder {(folder && folder.name) || "No folder"}</h2>
			</div>
		);
	}
}

export default withRouter(NoteSidebar);
