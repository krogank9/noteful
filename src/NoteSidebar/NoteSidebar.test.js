import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteSidebar from './NoteSidebar'

describe(`NoteSidebar component`, () => {
	const props = {
		folder: {
			"id": "1",
			"name": "test note folder name"
		}
	}

	it('renders with no props', () => {
		const wrapper = shallow(<NoteSidebar />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders the note sidebar with folder name given props', () => {
		const wrapper = shallow(<NoteSidebar {...props} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
