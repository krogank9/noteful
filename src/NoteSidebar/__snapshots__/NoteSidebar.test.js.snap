// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSidebar component renders the note sidebar with folder name given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`NoteSidebar component renders with no props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
